inlang/source-code/sdk2/src/lix-plugin/reportConflicts.ts

```diff
@@ -0,0 +1,53 @@
+import type { Conflict, LixPlugin } from "@lix-js/sdk";
+import { getChangesNotInTarget } from "./utilities/getChangesNotInTarget.js";
+import { getFirstCommonParent } from "./utilities/getFirstCommonParent.js";
+import { getLastChildOfChange } from "./utilities/getLastChildOfChange.js";
+
+export const reportConflicts: LixPlugin["reportConflicts"] = async ({
+	sourceLix,
+	targetLix,
+}) => {
+	const result: Conflict[] = [];
+	const changesNotInTarget = await getChangesNotInTarget({
+		sourceLix,
+		targetLix,
+	});
```

Comment:
Letting a plugin deal with all logic on how to derive conflicts likely reduced the risk atm of building wrong abstractions that don't generalize. 

See https://github.com/opral/lix-sdk/issues/42 as an example. A CSV plugin would need to capture dependencies between changes and retrieve / dynamically compute snapshots based on the dependencies.